jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -101,40 +100,76 @@ def _mode_selected_changed(self, event={}):
+
+        if subset_definition is not None and subset_definition not in self.subset_definitions:
+            self.subset_definitions.append(subset_definition)
+            self.subset_types.append(subset_type)
+            print(self.subset_types)
+
+    def _get_subset_definition(self, *args):
+        """
+        Retrieve the parameters defining the selected subset, for example the
+        upper and lower bounds for a simple spectral subset.
+        """
+        self.subset_definitions = []
+        self.subset_types = []
         subset_group = [s for s in self.app.data_collection.subset_groups if
                         s.label == self.subset_selected][0]
         subset_state = subset_group.subset_state
-        subset_class = subset_state.__class__
 
-        if subset_class in (OrState, AndState, XorState, InvertState):
-            self.subset_classname = "Compound Subset"
-            self.has_subset_details = False
-        else:
-            if isinstance(subset_state, RoiSubsetState):
-                self.subset_classname = subset_state.roi.__class__.__name__
-                self.has_subset_details = True
-                if self.subset_classname == "CircularROI":
-                    x, y = subset_state.roi.get_center()
-                    self.subset_definition = {"X Center": x,
-                                              "Y Center": y,
-                                              "Radius": subset_state.roi.radius}
-                elif self.subset_classname == "RectangularROI":
-                    temp_def = {}
-                    for att in ("Xmin", "Xmax", "Ymin", "Ymax"):
-                        temp_def[att] = getattr(subset_state.roi, att.lower())
-                    self.subset_definition = temp_def
-                elif self.subset_classname == "EllipticalROI":
-                    self.subset_definition = {"X Center": subset_state.roi.xc,
-                                              "Y Center": subset_state.roi.yc,
-                                              "X Radius": subset_state.roi.radius_x,
-                                              "Y Radius": subset_state.roi.radius_y}
-            elif isinstance(subset_state, RangeSubsetState):
-                self.subset_classname = "Range"
-                self.subset_definition = {"Upper bound": subset_state.hi,
-                                          "Lower bound": subset_state.lo}
-                self.has_subset_details = True
-            else:
-                self.subset_classname = subset_class.__name__
-                self.has_subset_details = False
+        self._unpack_nested_subset(subset_state)
+
+        # Trick traitlets into updating the displayed values
+        subset_definitions = self.subset_definitions
+        self.subset_definitions = []
+        self.subset_definitions = subset_definitions
+        subset_types = self.subset_types
+        self.subset_types = []
+        self.subset_types = subset_types
```

Comment:
I'm surprised this is needed since you set them to empty at the beginning of the method.